### PR TITLE
PT-3927: Push code should handle usernames with extra leading and/or trailing spaces correctly 

### PR DESCRIPTION
**Pull request by allasm**

### CODE REVIEW

File: components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java [hunk 1]
```diff
@@ -382,44 +394,113 @@ protected boolean userTokensEnabled(BaseObject serverConfig)
...
+
+        public boolean isSuccessful() {
+            return this.authorizedUsername != null;
+        }
+
+        public boolean isFailed() {
+            return !this.isSuccessful();
+        }
+
+        public String getAuthorizedusername() {
```
Comment (marta-) [inline components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java]: Please standardize this method's name:
```suggestion
        public String getAuthorizedUsername() {
```

File: components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java [hunk 2]
```diff
@@ -457,12 +540,12 @@ public JSONObject receivePatient()
...
-            JSONObject loginError = validateLogin(request, context);
-            if (loginError != null) {
-                return loginError;
+            LoginResult loginResult = validateLogin(request, context);
+            if (loginResult.isFailed()) {
+                return loginResult.getFailedLoginDetails();
             }
 
-            String userName = request.getParameter(ShareProtocol.CLIENT_POST_KEY_NAME_USERNAME);
+            String userName = loginResult.getAuthorizedusername();
```
Comment (sdumitriu) [inline components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java]:
```suggestion
            String userName = loginResult.getAuthorizedUsername();
```

File: components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java [hunk 3]
```diff
@@ -628,12 +711,12 @@ public JSONObject getConfiguration()
...
```
Comment (sdumitriu) [inline components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java]:
```suggestion
            String userName = loginResult.getAuthorizedUsername();
```

File: components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java [hunk 4]
```diff
@@ -684,12 +766,12 @@ public JSONObject getPatientURL()
...
```
Comment (sdumitriu) [inline components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java]:
```suggestion
            String userName = loginResult.getAuthorizedUsername();
```

**Pull request merged**